test/test_vector.py

```diff
@@ -0,0 +1,158 @@
+        can.interfaces.vector.canlib.vxlapi.xlGetChannelIndex = Mock(return_value=0)
+        can.interfaces.vector.canlib.vxlapi.XLuint64 = Mock(
+            return_value=ctypes.c_int64(0)
+        )
+        can.interfaces.vector.canlib.vxlapi.xlCanReceive = Mock(return_value=0)
+        can.interfaces.vector.canlib.vxlapi.xlReceive = Mock(return_value=0)
+        can.interfaces.vector.canlib.vxlapi.xlCanFdSetConfiguration = Mock()
+        can.interfaces.vector.canlib.vxlapi.xlCanSetChannelBitrate = Mock()
+        can.interfaces.vector.canlib.vxlapi.XL_CAN_EXT_MSG_ID = 0x80000000
+        can.interfaces.vector.canlib.vxlapi.XLevent = MagicMock()
+        can.interfaces.vector.canlib.vxlapi.XLcanTxEvent = MagicMock()
+        can.interfaces.vector.canlib.vxlapi.xlCanTransmit = Mock()
+        can.interfaces.vector.canlib.vxlapi.xlCanTransmitEx = Mock()
+        can.interfaces.vector.canlib.vxlapi.xlCanFlushTransmitQueue = Mock()
+        can.interfaces.vector.canlib.vxlapi.xlDeactivateChannel = Mock()
+        can.interfaces.vector.canlib.vxlapi.xlClosePort = Mock()
+        can.interfaces.vector.canlib.vxlapi.xlCloseDriver = Mock()
+        can.interfaces.vector.canlib.vxlapi.xlActivateChannel = Mock()
+        can.interfaces.vector.canlib.WaitForSingleObject = Mock()
+        self.bus = None
+
+    def tearDown(self) -> None:
+        if self.bus:
+            self.bus.shutdown()
+            self.bus = None
+
+    def test_bus_creation(self) -> None:
+        self.bus = can.Bus(channel=0, bustype="vector")
+        self.assertIsInstance(self.bus, canlib.VectorBus)
+        can.interfaces.vector.canlib.vxlapi.xlOpenDriver.assert_called()
+        can.interfaces.vector.canlib.vxlapi.xlGetApplConfig.assert_called()
+        can.interfaces.vector.canlib.vxlapi.xlOpenPort.assert_called()
+        can.interfaces.vector.canlib.vxlapi.xlCanFdSetConfiguration.assert_not_called()
+        can.interfaces.vector.canlib.vxlapi.xlCanSetChannelBitrate.assert_not_called()
+
+    def test_bus_creation_bitrate(self) -> None:
+        self.bus = can.Bus(channel=0, bustype="vector", bitrate="500000")
+        self.assertIsInstance(self.bus, canlib.VectorBus)
+        can.interfaces.vector.canlib.vxlapi.xlOpenDriver.assert_called()
+        can.interfaces.vector.canlib.vxlapi.xlGetApplConfig.assert_called()
+        can.interfaces.vector.canlib.vxlapi.xlOpenPort.assert_called()
+        can.interfaces.vector.canlib.vxlapi.xlCanFdSetConfiguration.assert_not_called()
+        can.interfaces.vector.canlib.vxlapi.xlCanSetChannelBitrate.assert_called()
+
+    def test_bus_creation_fd(self) -> None:
+        self.bus = can.Bus(channel=0, bustype="vector", fd=True)
+        self.assertIsInstance(self.bus, canlib.VectorBus)
+        can.interfaces.vector.canlib.vxlapi.xlOpenDriver.assert_called()
+        can.interfaces.vector.canlib.vxlapi.xlGetApplConfig.assert_called()
+        can.interfaces.vector.canlib.vxlapi.xlOpenPort.assert_called()
```

Comment:
Looking through the code, it seems like in certain cases, the same Vector API is called, but with a different parameter to distinguish between various cases (eg. for FD vs. non-FD). Like here, the difference is:

```python
        if fd:
            vxlapi.xlOpenPort(
                self.port_handle,
                self._app_name,
                self.mask,
                permission_mask,
                rx_queue_size,
                vxlapi.XL_INTERFACE_VERSION_V4, # <--
                vxlapi.XL_BUS_TYPE_CAN,         # <--
            )
        else:
            vxlapi.xlOpenPort(
                self.port_handle,
                self._app_name,
                self.mask,
                permission_mask,
                rx_queue_size,
                vxlapi.XL_INTERFACE_VERSION,    # <--
                vxlapi.XL_BUS_TYPE_CAN,         # <--
            )
```

Does it make sense to also assert on those arguments for those situations, rather than just that the function was called?

Reply:
done :)